client/client.go

```diff
@@ -35,6 +35,27 @@ type Client struct {
     slotsLock *sync.Mutex
 }
 
+type MessageBody struct {
+    Method  string
+    Args    []interface{}
+}
+
+func (m *MessageBody) Marshal() ([]byte, error){
+    // To not call directly json.Marshal(MessageBody),
+    // maybe some fields in future cannot be exposed, and need specific treatments.
+    type _messageBody struct {
+        Method  string          `json:"method"`
+        Args    []interface{}   `json:"args"`
+    }
```

Comment:
IMHO If messageBody was private, it wouldn't be necessary to make the code more complex due to possible future requirements.